lib/validate.py

```diff
@@ -67,47 +94,83 @@ def box_plot_for_degree(log_directory, output_directory, channel_name):
             saver.save_csv([in_degree_fit_parameters[:, para_ind].tolist()], output_directory, "in_degree_"+str(parameters[para_ind])+"_2013_"+channel_name_iter[0]+"_cut_"+str(cutoff))
             saver.save_csv([total_degree_fit_parameters[:, para_ind].tolist()], output_directory, "total_degree_"+str(parameters[para_ind])+"_2013_"+channel_name_iter[0]+"_cut_"+str(cutoff))
 
-def keywords_hits_overlap(log_directory, output_directory, channel_name):
-    # Correlational: overlap for keyword digest and HITS
-    for month in xrange(1, 13):
-        log_data_m1 = reader.linux_input(log_directory, channel_name, "2013-"+str(month)+"-1", "2013-"+str(month)+"-31")
+def keywords_hits_overlap(log_directory, output_directory, channel_name, start_date, end_date):
+    """
+        The function iterates through the months in the given date range and produces the authorities, top keywords and
+        top hubs for the current month and the next month.
```

Comment:
The function also checks for the percentage overlap between the list of Hubs generated for two consecutive months. Similarly check is made for Authorities and Keywords.

Reply:
done